src/escalation/jira_escalation.py

```diff
@@ -0,0 +1,226 @@
+import datetime
+from typing import Optional
+
+# import click
+# from jinja2 import Environment
+# from jinja2 import FileSystemLoader
+from simple_logger.logger import get_logger
+
+# from src.escalation.send_slack_notification import send_slack_notification
+from src.objects.jira_base import Jira
+from src.objects.slack_base import SlackClient
+
+
+LOGGER = get_logger(name=__name__)
+
+
+class Jira_Escalation:
+    """ """
+
+    def __init__(
+        self,
+        jira: Jira,
+        slack_client: SlackClient,
+        slack_channel: str,
+        default_labels: list[str],
+        additional_labels: list[str],
+        default_jira_project: str,
+    ):
+        self.default_labels = default_labels
+        self.additional_labels = additional_labels
+        self.default_project = default_jira_project
+        self.slack_client = slack_client
+        self.jira = jira
+
+        # build query and process ocp lp firewatch tickets in NOACK and ACK status
+        jira_query = ""
+        for status in ["NO ACK", "ACK"]:
+            if default_jira_project:
+                jira_query = self.build_jql(status)
+            jira_query = self.build_jql(status)
+
+            LOGGER.info(f" jira-query: {jira_query}")
+
+            self.process_issues(jira_query, slack_channel)
+
+        # process PQE tickets
+        pqe_jira_query = (
+            f"Project != {default_jira_project} AND status not in('Resolved','Blocked','Closed','Backlog','Done')"
+        )
+
+        if self.additional_labels:
+            formatted_additional_labels = '","'.join(self.additional_labels)
+            LOGGER.info(f" add labels {formatted_additional_labels}")
+            pqe_jira_query += f' AND (labels IN("{formatted_additional_labels}"))'
+
+        if self.default_labels:
+            default_condition = " AND".join([f'labels = "{label}"' for label in self.default_labels])
+            pqe_jira_query += f" AND ({default_condition})"
+
+        LOGGER.info(f" jira query PQE tickets: {pqe_jira_query}")
+        self.process_issues(pqe_jira_query, slack_channel)
+
+    def process_issues(self, jira_query: str, slack_channel: str) -> None:
+        """ 
+        Process jira issues for escalation based on escaltion criteria
+        Args:
+          jira_query (string): JQL query string to get the issues
+          slack_channel (str): slack channel name to send notifications
+        """
```

Comment:
I think we can pass the user or email as function parameter here.  Maybe for now its fine for testing, but you can move the hardcoded test email outside function into a variable

Reply:
removed hard code parameters